google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/schema/SpannerTableStatements.java

```diff
@@ -53,17 +57,38 @@ public SpannerTableStatements(SpannerDialect spannerDialect) {
    */
   public List<String> dropTable(Table table, Metadata metadata) {
     ArrayList<String> dropStrings = new ArrayList<>();
-
-    Iterator<Index> iteratorIdx = table.getIndexIterator();
-    while (iteratorIdx.hasNext()) {
-      Index curr = iteratorIdx.next();
-      dropStrings.add("drop index " + curr.getName());
+    for (String indexName : getTableIndices(table)) {
+      dropStrings.add("drop index " + indexName);
     }
 
     dropStrings.add(this.spannerDialect.getDropTableString(table.getQuotedName()));
     return dropStrings;
   }
 
+  private Set<String> getTableIndices(Table table) {
```

Comment:
Do we have test coverage for all 3 types of sources of indices?

Reply:
Yes! After addressing Elena's comment there are only two sources of indices we need to address:
- "unique indices" - created by the `@Column(unique = true)`; the unit tests added verify this.
- "normal indices" - created by using `@Index(...)`. This is present on the `Employee` entity and is verified by pre-existing tests.